stablehlo/dialect/Version.cpp

```diff
@@ -83,9 +133,9 @@ Version Version::fromCompatibilityRequirement(
     case CompatibilityRequirement::NONE:
       return Version::getCurrentVersion();
     case CompatibilityRequirement::WEEK_4:
-      return Version(1, 7, 0);  // v1.7.0 - Sept 05, 2024
+      return getVersionTaggedOnOrAfterDays(28);
```

Comment:
How about hardcoding `4 * 7` and `12 * 7`. Compiler would hopefully take care of the folding and it's a bit more readable.